src/components/forms/CreateActionItemForm.tsx

```diff
@@ -0,0 +1,94 @@
+import { Component, Setter } from 'solid-js'
+import { Form, FormType } from 'solid-js-form'
+import * as Yup from 'yup'
+import { createMutation, createQuery } from 'solid-urql'
+import { useParams } from 'solid-app-router'
+
+import Button from '../Button'
+import Input from '../Input'
+import FormDropdown from '../FormDropdown'
+import { User } from '../../types/user'
+
+const GET_USERS = `
+  query {
+    users {
+      id
+      firstName
+    }
+  }
+`
+
+const CREATE_ACTION_ITEM = `
+  mutation ($input: CreateActionItemInput!) {
+    createActionItem(createActionItemInput: $input){
+      id
+    }
+  }
+`
+
+type Props = {
+  setShouldDisplay: Setter<Boolean>
+}
+
+const CreateActionItemsForm: Component<Props> = ({ setShouldDisplay }) => {
+  const [usersResult] = createQuery({ query: GET_USERS })
+
+  const userOptions = () =>
+    usersResult()?.users.map(({ id, firstName }: User) => ({
+      id,
+      label: firstName,
+    }))
+
+  const [_, createActionItem] = createMutation(CREATE_ACTION_ITEM)
+  const { id: incidentId } = useParams()
+
+  const handleOnSubmit = async (
+    form: FormType.Context<{
+      text: string
+      ownerId: string
+      incidentId: string
+    }>
+  ) => {
+    const variables = {
+      input: {
+        text: form.values.text,
+        ownerId: form.values.ownerId,
+        incidentId,
```

Comment:
I'm surprised this works since it seems like it is coming from the `form` which is `''`.